doc/mgmt/SONiC_Design_Doc_Unified_FRR_Mgmt_Interface.md

```diff
@@ -0,0 +1,1069 @@
+
+```
+sonic# show ipv6 prefix-list
+IPv6 prefix list pref2:
+     permit 2222::/64 ge 65 le 65
+     permit 2223::/64 ge 65 le 128
+sonic#
+```
+|Command Description|CLI Command      |
+|:------------------|:-----------------|
+| Display BGP community list information | show bgp community-list |
+
+```
+sonic# show bgp community-list
+Standard community list com1:  match: ANY
+     local-AS
+Expanded community list com2:   match: ANY
+     Extended1
+sonic#
+```
+|Command Description|CLI Command      |
+|:------------------|:-----------------|
+| Display BGP extended community list information | show bgp ext-community-list |
+
+```
+sonic# show bgp ext-community-list
+Standard extended community list extcom1:  match: ANY
+     rt:2:2
+Expanded extended community list extcom2:  match: ANY
+     extcom
+sonic#
+```
+
+|Command Description|CLI Command      |
+|:------------------|:-----------------|
+| Display BGP AS-Path list information | show bgp as-path-access-list |
+
+```
+sonic# show bgp as-path-access-list
+AS path list aspath1:
+   members: 1:1+
+sonic#
+```
+
+
+#### 3.6.2.3 Debug Commands
+N/A
+
+#### 3.6.2.4 IS-CLI Compliance
+NA
+### 3.6.3 REST API Support
+Various REST operations (POST/PUT/PATCH/GET/DELETE) are supported for the BGP and route map configurations.
+# 4 Flow Diagrams
+
+## 4.1 Configuration Sequence
+
+![FRR-BGP-CONFIG-SEQUENCE](images/frr-bgp-config-sequence.jpg)
+
+## 4.2 CLI Show Command Sequence
+
+### 4.2.1 CLI Show Sequence - Config information
+![FRR-BGP-CLI-SHOW-SEQUENCE1](images/frr-bgp-cli-show-sequence11.jpg)
+
+### 4.2.2 CLI Show Sequence - State/Statistics
+![FRR-BGP-CLI-SHOW-SEQUENCE2](images/frr-bgp-cli-show-sequence22.jpg)
+
+## 4.3 REST Get Sequence
+
+### 4.3.1 REST Get Sequence - Config information
+![FRR-BGP-REST-GET-SEQUENCE1](images/frr-bgp-rest-get-sequence1.jpg)
+
+### 4.3.2 REST Get Sequence - State/Statistics
+![FRR-BGP-REST-GET-SEQUENCE2](images/frr-bgp-rest-get-sequence2.jpg)
+
+# 5 Error Handling
+
+The complete validation of BGP parameters is performed in the SONiC management framework based on open config BGP/Routing policy YANGs and against corresponding SONiC YANGs. Appropriate error code is returned for invalid configurations or on failures due to a dependency from management framework and bgpcfgd modules and the same is logged into the syslog, it is not expected for bgpcfgd (FRR CLI command execution) to return validation errors. However, such errors are logged into syslog for debugging purposes.
+
+# 6 Serviceability and Debug
+
+  1. Tables added to CONFIG_DB for FRR-BGP unified management are accessible via table dump.
+  2. Syslog messages are added at appropriate places to help trace a failure.
+  3. Leverages existing debug mechanism & framework if any.
+
+# 7 Warm Boot Support
+
+This enhancement to FRR-BGP Unified management framework does not disrupt data plane traffic during warmboot. No special handling required for warmboot.
+
+# 8 Scalability
+
+# 9 Unit Test
```

Comment:
please enable the testing part. the description of test is not enough.

Reply:
Sure, will add the UT scripts in the code.